velox/common/memory/MemoryAllocator.cpp

```diff
@@ -161,26 +161,56 @@ bool MemoryAllocator::allocateNonContiguous(
     Allocation& out,
     ReservationCallback reservationCB,
     MachinePageCount minSizeClass) {
+  const MachinePageCount numPagesToFree = out.numPages();
+  const auto bytesToFree = AllocationTraits::pageBytes(numPagesToFree);
+  if (numPages == 0) {
+    if (bytesToFree > 0) {
```

Comment:
Can we use the same approach as cleanupCollateralAndReleaseReservation to handle the cleanup?